### DISCUSSION TITLE
Option to merge statusline and command line into a single line

### DISCUSSION THREAD

**Discussion by nik-rev** [helix-editor/helix#12016] [Ideas]
This is currently how the bottom of my editor looks:

![image](https://github.com/user-attachments/assets/6cd7b3ed-26a2-4271-969b-d9330a11793c)

So there is an entire line of whitespace that could be used by actual text instead

Can there be an option, like `editor.statusline.merge` (boolean) which will hide the statusline when there is a message ? so instead of this:

![image](https://github.com/user-attachments/assets/cdf10784-8ea9-45d2-ae6c-6c4285990465)

The error message will instead replace any text on the status line while it's shown

And when we type `:` command, it will also replace the status line

**Comment by diocletiann** [comment]
I would love this. Neovim has this option and it works well.

**Comment by nik-rev** [comment]
somewhat related: https://github.com/helix-editor/helix/pull/12020

**Comment by LechintanTudor** [comment]
I would like to have this option. This is how the statusline in Tmux works.

**Comment by erasin** [comment]
Hide Commandline #11223
feat: new statusline.merge-with-commandline option #12204
